sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -450,34 +451,24 @@ private Object getValueFromJson(String propertyName, Method method) {
    */
   @SuppressWarnings({"unchecked", "rawtypes"})
   private Object getDefault(PipelineOptions proxy, Method method) {
+    if (method.getReturnType().equals(ValueProvider.class)) {
```

Comment:
What happens if the user explicitly puts `RuntimeValueProvider` or `StaticValueProvider` as the getter return type? Can we (should we) throw?